[KamJiaYue] iP

KamJiaYue opened the pull request:
### **DUKE CHAT BOT**

> "Stay Hungry. Stay Foolish" - Steve Jobs ([source](https://www.youtube.com/watch?v=8Zx04h24uBs))

The Duke Chat Bot is a **FRIEND** of ours 😁😁!
It helps us remember **IMPORTANT** tasks like:

- To do's 
- Deadlines
- Events

You just have to:

1. Download it from [HERE](https://github.com/KamJiaYue/ip/releases/tag/v0.1)
2. ~~Run it~~ **WAKE IT UP** 
4. Start _Chatting_ and adding your tasks!

Features:

- [ ] Remember tasks
- [ ] Reminders (Coming Soon)

For Java Programmers, DO THIS in your `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Duke bot = new Duke("YOUR_BOT_NAME");
        bot.startUp("./data/duke.txt");
        bot.startService();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jyue487 committed:
Level-0

KamJiaYue committed:
level0

KamJiaYue committed:
level 1

KamJiaYue committed:
level 2

KamJiaYue committed:
level 3

KamJiaYue committed:
level 4

KamJiaYue committed:
A-TextUiTesting

KamJiaYue committed:
level 5

KamJiaYue committed:
level 6

KamJiaYue committed:
ATextUiTesting

KamJiaYue committed:
level 5

KamJiaYue committed:
A-Enums

KamJiaYue committed:
lastest

KamJiaYue committed:
Implement TaskList class, add save and load feature

KamJiaYue committed:
Implement TaskList class, add save and load feature

KamJiaYue committed:
Add TaskList class, implement save and load feature

KamJiaYue committed:
delete one line of extra space

KamJiaYue committed:
Merge branch 'branch-Level-7'

KamJiaYue committed:
Delete another line of extra space

KamJiaYue committed:
Merge branch-Level-7 without fast forward

KamJiaYue committed:
Add time formats specifically to Todo and Event Tasks

KamJiaYue committed:
Merge branch-Level-8 without fast forward

KamJiaYue committed:
Add Ui class

KamJiaYue committed:
Complete Ui Implementation with manual testing

KamJiaYue committed:
Complete Storage Implementation with manual testing

KamJiaYue committed:
Complete TaskList Implementation with manual testing

KamJiaYue committed:
Complete Implementation of Parser with manual testing

KamJiaYue committed:
Merge with branch-A-MoreOOP

KamJiaYue committed:
Add package

KamJiaYue committed:
Merge branch-A-Packages with no fast forward

KamJiaYue committed:
Merge branch 'add-gradle-support' of ssh://github.com/KamJiaYue/ip into branch-A-Gradle
This merge is necessary for adding Gradle into my project

KamJiaYue committed:
Add Gradle

KamJiaYue committed:
Merge branch-A-Gradle without fast forward

KamJiaYue committed:
Change main class from seedu.duke.Duke to duke.Duke

KamJiaYue committed:
Merge branch-A-Gradle without fast forward

KamJiaYue committed:
Add JUnit Tests

KamJiaYue committed:
Merge branch-A-JUnit without fast forward

KamJiaYue committed:
Add .jar file

KamJiaYue committed:
Merge branch-A-Jar without fast forward

KamJiaYue committed:
Complete Implementation of Find

KamJiaYue committed:
Correct coding style

KamJiaYue committed:
Write JavaDoc

KamJiaYue committed:
Add {@inheritDoc}

KamJiaYue committed:
Merge branch-A-JavaDoc without fast forward

KamJiaYue committed:
Merge branch-A-CodingStandard without fast forward

KamJiaYue committed:
Merge branch-Level-9 without fast forward

KamJiaYue committed:
Add {@inheritDoc]

KamJiaYue committed:
Check coding style

Daphne789 commented on src/main/java/seedu/duke/Ui.java:
```diff
@@ -11,10 +11,10 @@
-    public static final String i5 = Ui.i4 + " ";
-    public static final String i7 = Ui.i5 + "  ";
-    public static final String line = Ui.i4 + "——————————————————————————————————————————————————————————————————";
+    public static final String I4 = "    ";
+    public static final String I5 = Ui.I4 + " ";
+    public static final String I7 = Ui.I5 + "  ";
```
```suggestion
    public static final String INDENT_4 = "    ";
    public static final String INDENT_5 = INDENT_4 + " ";
    public static final String INDENT_7 = INDENT_5 +  "  ";`
```
I like how you make the indents into constants so it will easily be managed, however perhaps can consider a more intuitive constant name?

Daphne789 commented on src/main/java/seedu/duke/Ui.java:
```diff
@@ -36,25 +36,29 @@ public Ui(String name) {
      */
     public static void line() {
-        System.out.println(Ui.line);
+        System.out.println(Ui.LINE);
```
Is it necessary to use references for calling constants in the class itself?

Daphne789 commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -55,19 +55,19 @@ public Commands parse(String input) {
         }
 
         try {
             return Integer.parseInt(input.substring(5)) - 1;
         } catch (NumberFormatException e) {
-            throw new DukeException(Ui.i5 + "☹ OOPS!!! Please specify a valid task number to be marked.");
+            throw new DukeException(Ui.I5 + "☹ OOPS!!! Please specify a valid task number to be marked.");
         }
```
Any reason why you added the _try-catch block_ in the code even though you have _if statements_ that have handled several exceptions?

Daphne789 reviewed (commented):
I found your code pretty neat. I noted a minor coding standard violation in the constant name. Some redundancies make several parts of the code not easy to read. 
Overall, Good Job!

Daphne789 commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -59,21 +59,21 @@ public void startService() throws DukeException {
                 int unmarkIndex = this.ui.unmark();
                 if (taskList.isOutOfRange(unmarkIndex)) {
-                    throw new DukeException(Ui.i5 + "☹ OOPS!!! Please specify a valid task number to be unmarked.");
+                    throw new DukeException(Ui.I5 + "☹ OOPS!!! Please specify a valid task number to be unmarked.");
```
```suggestion
                    throw new DukeException(Ui.I5
                        + "☹ OOPS!!! Please specify a valid task number to be unmarked.");
```
Indentation for wrapped lines should be 8 spaces (i.e. twice the normal indentation of 4 spaces) more than the parent line.

Daphne789 reviewed (commented):
I found minor coding standard violations, especially for the layout: indentation for wrapped lines.

KamJiaYue committed:
Complete CheckStyle Implementation

KamJiaYue committed:
Merge branch-A-CheckStyle without fast forward

KamJiaYue committed:
Initialize JavaFX

KamJiaYue committed:
Add Anchor Pane

KamJiaYue committed:
Complete JavaFX Implementation

KamJiaYue committed:
Add GUI

KamJiaYue committed:
Merge branch-Level-10 without fast forward

KamJiaYue committed:
Add Varargs

KamJiaYue committed:
Add Varargs

KamJiaYue committed:
Merge branch-A-Varargs without fast forward

KamJiaYue committed:
Merge master branch

KamJiaYue committed:
Complete Stretch Goal

KamJiaYue committed:
Merge branch-A-MoreOOP without fast forward

KamJiaYue committed:
Add design and style

KamJiaYue committed:
Add styling

KamJiaYue committed:
Merge branch-A-Styling without fast forward

KamJiaYue committed:
Remove all indents in every output

KamJiaYue committed:
Remove all indents in every output

KamJiaYue committed:
Remove all indents in every output

Remove indentations for every system output

System outputs have indents.

The indents have made the output seem weird.

Deleting the indents produce a cleaner and
nicer output.

Let's remove unnecessary indentations
from the Duke, Ui and Parser classes.

KamJiaYue committed:
Merge branch 'master' of ssh://github.com/KamJiaYue/ip

KamJiaYue committed:
Remove all indents in every output

System output have indents.

The indents of the output have made
the output seemed weird.

Removing the indentations produces
a clean and nice output.

Let's delete the indentations from
the Duke, Ui and Parser class.

KamJiaYue committed:
Test commit message

Message

KamJiaYue committed:
Test commit message 2

Message 2

KamJiaYue committed:
Remove all indents in every output

System output have indents.

The indents of the output have made
the output seemed weird.

Removing the indentations produces
a clean and nice output.

Let's delete the indentations from
the Duke, Ui and Parser class.

KamJiaYue committed:
Remove all indents in every output

System output have indents.

The indents of the output have made
the output seemed weird.

Removing the indentations produces
a clean and nice output.

Let's delete the indentations from
the Duke, Ui and Parser class.

KamJiaYue committed:
Add some number of assertions at task and command package

Classes in task and command package (e.g. AddCommand,
DeleteCommand, Deadline, Event etc.) lack assertions.

There might be some programmer errors not properly
handled.

Adding some assertions to classes in task and command
package can handle potential programmer errors.

Let's insert assertions to classes in task and command
package.

Using java assertions is preferable to handle programmer
errors in this scenario.

Refer to this documentation while writing Java assertions
https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/errorHandling.html#assertions

KamJiaYue committed:
Complete Stream Implementation

KamJiaYue committed:
Improve Code Quality

KamJiaYue committed:
Improve Code Quality

KamJiaYue committed:
Merge pull request #1 from KamJiaYue/branch-A-Assertions

Add some number of assertions at task and command package

KamJiaYue committed:
Merge branch 'master' into branch-A-CodeQuality

Merge master branch to branch-A-CodeQuality branch

KamJiaYue committed:
Merge pull request #2 from KamJiaYue/branch-A-CodeQuality

Improve code quality

KamJiaYue committed:
Complete BCD-Extension C : BetterSearch

KamJiaYue committed:
Add .yml file for CI

KamJiaYue committed:
Test CI

KamJiaYue committed:
Test CI 2

KamJiaYue committed:
Test CI 3

KamJiaYue committed:
Test CI 4

KamJiaYue committed:
Add UG

KamJiaYue committed:
Add user guide

KamJiaYue committed:
Handle event time frame errors

KamJiaYue committed:
Merge pull request #6 from KamJiaYue/branch-handleEventTimeFrameError

Handle event time frame errors
